### ISSUE TITLE
Compilation failure on standard Keras MLP

### ISSUE THREAD

**Issue by omdv** [plaidml/plaidml#150]
Hello,

First my versions: `plaidml                           0.3.4
plaidml-keras                     0.3.4
Keras                             2.0.8`

Backend: `INFO:plaidml:Opening device "metal_intel(r)_iris(tm)_graphics_6100.0`

Background: I used to use plaidml for a while and it worked great. Recently I had to test few algorithms using MNIST and got upon a bug, so I decided to validate it on [official keras MNIST mlp example](https://github.com/keras-team/keras/blob/master/examples/mnist_mlp.py).

Bug
```
Train on 60000 samples, validate on 10000 samples
Epoch 1/20
ERROR:plaidml:Compiler::Build> Compilation failure:
Compilation failed:

program_source:79:7: warning: unused variable 'tid'
  int tid = _tid;
      ^
program_source:495:11: warning: unused variable 'x0'
      int x0 = ((64 * x0_lid) + x0_tid);
          ^
program_source:2760:7: warning: unused variable 'tid'
  int tid = _tid;
      ^
program_source:3095:12: error: cannot initialize a variable of type 'int2' (aka 'vector_int2') with an rvalue of type 'vec<char, 2>' (vector of 2 'char' values)
      int2 LX_T168 = select((char2)((float2)0), (char2)((float2)1), ((bool2)LX_T167));
           ^         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Based on the description I suspected an issue with input data, but I don't see any issues. Topology is quite standard - it is a basic example.

Code
```
import plaidml.keras
plaidml.keras.install_backend()

import keras
from keras.datasets import mnist
from keras.models import Sequential
from keras.layers import Dense, Dropout
from keras.optimizers import RMSprop

batch_size = 128
num_classes = 10
epochs = 20

# the data, split between train and test sets
(x_train, y_train), (x_test, y_test) = mnist.load_data()

x_train = x_train.reshape(60000, 784)
x_test = x_test.reshape(10000, 784)
x_train = x_train.astype('float32')
x_test = x_test.astype('float32')
x_train /= 255
x_test /= 255
print(x_train.shape[0], 'train samples')
print(x_test.shape[0], 'test samples')

# convert class vectors to binary class matrices
y_train = keras.utils.to_categorical(y_train, num_classes)
y_test = keras.utils.to_categorical(y_test, num_classes)
y_train = y_train.astype('float32')
y_test = y_test.astype('float32')

model = Sequential()
model.add(Dense(512, activation='relu', input_shape=(784,)))
model.add(Dropout(0.2))
model.add(Dense(512, activation='relu'))
model.add(Dropout(0.2))
model.add(Dense(num_classes, activation='softmax'))

model.summary()

model.compile(loss='categorical_crossentropy',
              optimizer=RMSprop(),
              metrics=['accuracy'])

history = model.fit(x_train, y_train,
                    batch_size=batch_size,
                    epochs=epochs,
                    verbose=1,
                    validation_data=(x_test, y_test))
score = model.evaluate(x_test, y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])
```

Any suggestions will be appreciated. Thanks.

**Comment by omdv** [comment]
It is related to Metal backend, as switching to opencl removes the issue, but training diverges. I thought this may be related to [128](https://github.com/plaidml/plaidml/issues/128) but I changed all types to 32bit and also it was fixed in previous release. Any other suggestions?

**Comment by OliverUrbann** [comment]
I have the same issue with the same setup in the dense layers. However, switching to CPU or OpenCL does not fix it. Works only with tensorflow backend.

**Comment by brianretford** [comment]
Hey all, sorry we haven't paid any attention to this yet. Don't know if you saw the news: https://www.geekwire.com/2018/intel-acquires-seattle-based-deep-learning-startup-vertex-ai-bolster-artificial-intelligence-efforts/

It's taken us a bit of time to ramp up in the new environs, we hope to have more dev cycles to look into these issues in the nearish future.

**Comment by PeteSero** [comment]
@omdv: Try removing metrics=['accuracy'] from model.compile()

If that compiles successfully, you can try out other metrics; they are just informational after all.

**Comment by omdv** [comment]
@PeteSero this fixed it. Thanks! Other metrics, including custom work fine.
